process/smartContract/builtInFunctions/esdtTransfer.go

```diff
@@ -51,6 +51,11 @@ func NewESDTTransferFunc(
 	return e, nil
 }
 
+// SetNewGasConfig is called whenever gas cost is changed
+func (e *esdtTransfer) SetNewGasConfig(gasCost *process.GasCost) {
+	e.funcGasCost = gasCost.BuiltInCost.ESDTBurn
```

Comment:
We need to protect `e.funcGasCost` with a mutex

Reply:
no need. there is no possibility to call the from different go routines

Reply:
protected all the way